alburnum/maas/viscera/boot_sources.py

```diff
@@ -0,0 +1,74 @@
+"""Objects for boot sources."""
+
+__all__ = [
+    "BootSource",
+    "BootSources",
+]
+
+from . import (
+    check,
+    check_optional,
+    Object,
+    ObjectField,
+    ObjectSet,
+    ObjectType,
+    parse_timestamp,
+)
+
+
+class BootSourcesType(ObjectType):
+    """Metaclass for `BootSources`."""
+
+    def __iter__(cls):
+        return map(cls._object, cls._handler.read())
+
+    def create(cls, url, *, keyring_filename=None, keyring_data=None):
+        """Create a new `BootSource`."""
+        if keyring_filename is None and keyring_data is None:
+            raise ValueError(
+                "Either keyring_filename and keyring_data must be set.")
+        data = cls._handler.create(
+            url=url,
+            keyring_filename=(
+                "" if keyring_filename is None else keyring_filename),
+            keyring_data=(
+                "" if keyring_data is None else keyring_data))
+        return cls._object(data)
+
+
+class BootSources(ObjectSet, metaclass=BootSourcesType):
+    """The set of boot sources."""
+
+    @classmethod
+    def read(cls):
+        """Get list of `BootSource`'s."""
+        return cls(cls)
```

Comment:
I think if the method does not need access to the actual instance and can work as a class method then its best to place it there. So I am for keeping it on the meta-class.